core/sender/Sender.cpp

```diff
@@ -1928,6 +1928,55 @@ bool Sender::Send(const std::string& projectName,
         projectName, sourceId, logGroup, config, mergeType, logGroupSize, defaultRegion, filename, context);
 }
 
+int Sender::SendDirectly(std::vector<sls_logs::Log>& logs, Config* config) {
+    uint32_t nowTime = time(NULL);
+    const size_t maxCount = INT32_FLAG(merge_log_count_limit) / 4;
+    for (size_t beginIndex = 0; beginIndex < logs.size(); beginIndex += maxCount) {
+        size_t endIndex = beginIndex + maxCount;
+        if (endIndex > logs.size()) {
+            endIndex = logs.size();
+        }
+        sls_logs::LogGroup logGroup;
```

Comment:
why the logGroup is assembly repeatly?

Especially, when merge_log_count_limit > logs.size(), the loops are same.